exercises/dominoes/example.rb

```diff
@@ -0,0 +1,37 @@
+module Dominoes
+  def self.chain(dominoes)
+    return dominoes if dominoes.empty?
+
+    first = dominoes.first
+
+    subchain = try_subchain(dominoes.drop(1), *first)
+    subchain && [first] + subchain
+  end
+
+  def self.try_subchain(dominoes, chain_left, chain_right)
+    return chain_left == chain_right ? [] : nil if dominoes.empty?
+
+    dominoes.each_with_index { |domino, i|
+      other_dominoes = dominoes.take(i) + dominoes.drop(i + 1)
+      domino_left, domino_right = domino
+      if domino_left == chain_right
+        # Try to add domino (unflipped) to chain
+        if (subchain = try_subchain(other_dominoes, chain_left, domino_right))
+          return [domino] + subchain
+        end
+      elsif domino_right == chain_right
+        # Try to add domino (flipped) to chain
+        if (subchain = try_subchain(other_dominoes, chain_left, domino_left))
+          return [[domino_right, domino_left]] + subchain
+        end
+      end
+    }
+
+    # Found no suitable chain.
+    nil
```

Comment:
Sure. I'll note that that comment was partially targeting https://github.com/exercism/xruby/pull/469#discussion_r84790337 to explain why in this case it cannot be that this method always returns the same object - it has to sometimes return nil instead of list. It's useful to have the comment somewhere, so `example` it will be. Putting it in the test file didn't seem right since it wasn't relevant to all students.